bundles/org.openhab.binding.linky/src/main/java/org/openhab/binding/linky/internal/LinkyHandlerFactory.java

```diff
@@ -50,24 +62,52 @@
  * The {@link LinkyHandlerFactory} is responsible for creating things handlers.
  *
  * @author Gaël L'hopital - Initial contribution
+ * @author Laurent Arnal - Rewrite addon to use official dataconect API
  */
 @NonNullByDefault
 @Component(service = ThingHandlerFactory.class, configurationPid = "binding.linky")
-public class LinkyHandlerFactory extends BaseThingHandlerFactory {
+public class LinkyHandlerFactory extends BaseThingHandlerFactory implements LinkyAccountHandler {
     private static final DateTimeFormatter LINKY_FORMATTER = DateTimeFormatter.ofPattern("uuuu-MM-dd'T'HH:mm:ss.SSSX");
+    private static final DateTimeFormatter LINKY_LOCALDATE_FORMATTER = DateTimeFormatter.ofPattern("uuuu-MM-dd");
+    private static final DateTimeFormatter LINKY_LOCALDATETIME_FORMATTER = DateTimeFormatter
+            .ofPattern("uuuu-MM-dd HH:mm:ss");
     private static final int REQUEST_BUFFER_SIZE = 8000;
 
+    private final String clientId = "e551937c-5250-48bc-b4a6-2323af68db92";
```

Comment:
Is this supposed to remain ? Maybe move it to binding constants ?

Reply:
yes we need it for MyElectricalData authorization support.
I move it to Binding constants.